pkg/neobench/parser.go

```diff
@@ -34,12 +42,25 @@ func Parse(filename, script string, weight float64) (Script, error) {
 		return Script{}, c.err
 	}
 
-	return Script{
-		Name:     filename,
-		Readonly: false, // TODO
-		Commands: commands,
-		Weight:   weight,
-	}, nil
+	output.Commands = commands
+	return output, nil
+}
+
+func metaMetaCommand(output Script, c *parseContext) *Script {
+	expect(c, '#')
+	cmd := ident(c)
+
+	switch cmd {
+	case "autocommit":
+		output.Autocommit = true
+		return &output
+	case "readonly":
+		output.Readonly = true
```

Comment:
I think this is already automatically detected; neobench runs an EXPLAIN of the benchmark query before starting the benchmark, which picks up on syntax errors and whether the query is readonly.. though it has the drawback that it means you can't run readonly queries against the leader, if you wanted to force that.

Reply:
ah I saw the TODO and thought I was helping